csharp/src/Google.Protobuf/Reflection/EnumValueDescriptor.cs

```diff
@@ -85,6 +85,24 @@ public T GetOption<T>(Extension<EnumValueOptions, T> extension)
             return value is IDeepCloneable<T> ? (value as IDeepCloneable<T>).Clone() : value;
         }
 
+        /// <summary>
+        /// Gets a single value enum value option for this descriptor
+        /// </summary>
+        public bool TryGetOption<T>(Extension<EnumValueOptions, T> extension, out T value)
```

Comment:
Adding this method is basically just duplicating the Extension API and we should avoid that. 
See https://github.com/protocolbuffers/protobuf/pull/7434 and let me know what you think.